tests/RangeParserTest.php

```diff
@@ -0,0 +1,368 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Db\Pgsql\Tests;
+
+use Yiisoft\Db\Pgsql\RangeParser;
+use Yiisoft\Db\Pgsql\Schema;
+use DateInterval;
+use Yiisoft\Db\Query\Query;
+use Yiisoft\Db\Expression\Expression;
+
+final class RangeParserTest extends TestCase
+{
+    private const RANGES = [
+        1 => [
+            'int_range' => [1, 10],
+            'bigint_range' => ['2147483647', '2147483650'],
+            'num_range' => [10.5, 20.7],
+            'ts_range' => ['2017-10-20 10:10:00', '2018-10-20 15:10:00'],
+            'ts_tz_range' => ['2018-10-20 10:10:00+00:00', '2019-10-20 15:10:00+00:00'],
+            'date_range' => ['2020-12-01', '2021-01-01'],
+        ],
+        2 => [
+            'int_range' => [100, null],
+            'bigint_range' => ['4147483647', null],
+            'num_range' => [30.7, null],
+            'ts_range' => ['2017-10-20 10:10:00', null],
+            'ts_tz_range' => ['2018-10-20 10:10:00+00:00', null],
+            'date_range' => ['2020-12-01', null],
+        ],
+        3 => [
+            'int_range' => [null, 10],
+            'bigint_range' => [null, '2147483650'],
+            'num_range' => [null, 20.7],
+            'ts_range' => [null, '2018-10-20 15:10:00'],
+            'ts_tz_range' => [null, '2019-10-20 15:10:00+00:00'],
+            'date_range' => [null, '2021-01-01'],
+        ],
+    ];
+
+    public function testSimpleRanges(): void
+    {
```

Comment:
It's very complex tests. Please - use dataProvider and split to many small and readly tests